core/drivers/crypto/caam/hal/common/hal_cfg_dt.c

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2017-2019 NXP
+ *
+ * Brief   CAAM Configuration.
+ */
+#include <caam_common.h>
+#include <caam_hal_cfg.h>
+#include <caam_hal_jr.h>
+#include <caam_jr.h>
+#include <kernel/dt.h>
+#include <libfdt.h>
+
+static const char *dt_caam_match_table = {
+	"fsl,sec-v4.0",
+};
+
+static const char *dt_jr_match_table = {
+	"fsl,sec-v4.0-job-ring",
+};
+
+/*
+ * Finds the Job Ring reserved for the Secure Mode in the DTB
+ *
+ * @fdt         Reference to the Device Tree
+ * @status      Status mask flag of the node to found
+ * @find_node   [out] Node offset found
+ */
+static paddr_t find_jr_offset(void *fdt, int status, int *find_node)
+{
+	paddr_t jr_offset = 0;
+	int node = -FDT_ERR_NOTFOUND;
+
+	node = fdt_node_offset_by_compatible(fdt, 0, dt_jr_match_table);
+
+	while (node != -FDT_ERR_NOTFOUND) {
```

Comment:
For me this is clear like this and result is the same.

Reply:
It may be clear to you who might have written this and have become accustomed to funny looking loops. This is a suggestion which will make it easier for others to maintain the code you're upstreaming.

Reply:
if you want, I'll change it

Reply:
I guess it's safe to assume that @etienne-lms wants it, and I think it's an improvement.